database/user.go

```diff
@@ -225,25 +256,40 @@ func (db *DB) UserByStripeID(ctx context.Context, id string) (*User, error) {
 }
 
 // UserCreate creates a new user in the DB.
-func (db *DB) UserCreate(ctx context.Context, sub string, tier int) (*User, error) {
-	// Check for an existing user with this sub.
-	users, err := db.managedUsersByField(ctx, "sub", sub)
+func (db *DB) UserCreate(ctx context.Context, email, pass, sub string, tier int) (*User, error) {
+	// Check for an existing user with this email.
+	users, err := db.managedUsersByField(ctx, "email", email)
 	if err != nil && !errors.Contains(err, ErrUserNotFound) {
 		return nil, errors.AddContext(err, "failed to query DB")
 	}
 	if len(users) > 0 {
 		return nil, ErrUserAlreadyExists
 	}
-	_, email, err := jwt.UserDetailsFromJWT(ctx)
-	if err != nil {
-		// Log the error but don't do anything differently.
-		db.staticLogger.Debugf("We failed to extract the expected user infotmation from the JWT token. Error: %s", err.Error())
+	// Check for an existing user with this sub.
+	if sub != "" {
+		users, err := db.managedUsersByField(ctx, "sub", sub)
+		if err != nil && !errors.Contains(err, ErrUserNotFound) {
+			return nil, errors.AddContext(err, "failed to query DB")
+		}
+		if len(users) > 0 {
+			return nil, ErrUserAlreadyExists
+		}
+	} else {
```

Comment:
You can invert this can't you?
```
if sub == "" {
  sub = generateSub()
}
users, err := db.managedUsersByField(ctx, "sub", sub)
if err != nil && !errors.Contains(err, ErrUserNotFound) {
	return nil, errors.AddContext(err, "failed to query DB")
}
if len(users) > 0 {
	return nil, ErrUserAlreadyExists
}
```

Reply:
Yes, I can. I am not sure I really need to, though. Hear me out:

If there is a `sub` passed with the request then there's a good chance that the user is trying to log in or create a user based on an existing identity. That identity might already be registered in the DB, so we want to make sure it's not there before we create a new account with it.

Alternatively, if we're generating the `sub`, the chances for a collision between a new UUID and an existing one is close to zero. Close enough, actually, that we don't need to check - that's the entire premise of UUIDs. That's why I'm skipping the second trip to the DB.

Having in mind that this request doesn't need to be fast, I don't mind checking for a `sub` collision even when we generate it, so I'll just add it.